src/main/groovy/org/gradle/android/workarounds/room/JavaCompileWorkaround.groovy

```diff
@@ -0,0 +1,143 @@
+package org.gradle.android.workarounds.room
+
+import org.gradle.android.workarounds.RoomSchemaLocationWorkaround
+import org.gradle.android.workarounds.room.androidvariants.ConfigureVariants
+import org.gradle.android.workarounds.room.argumentprovider.JavaCompilerRoomSchemaLocationArgumentProvider
+import org.gradle.android.workarounds.room.argumentprovider.RoomSchemaLocationArgumentProvider
+import org.gradle.android.workarounds.room.task.RoomSchemaLocationMergeTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.execution.TaskExecutionGraph
+import org.gradle.api.file.Directory
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.process.CommandLineArgumentProvider
+
+class JavaCompileWorkaround extends AnnotationProcessorWorkaround<JavaCompilerRoomSchemaLocationArgumentProvider> {
```

Comment:
can we add a comment here to explain when this is used? like if is used ONLY when java is used without KAPT or if if it is used together with KAPT was well.